### PR TITLE
Check inputs for converting 3d surveys to 2d lines

### PR DESCRIPTION
**Pull request by santisoler**
#### Summary

Add checks to validate the `survey` and `lineID` inputs in the `convert_survey_3d_to_2d_lines` function. Add tests to check if errors are raised after passing invalid arguments. Improve the docstring of the function.

#### PR Checklist
* [ ] If this is a work in progress PR, set as a Draft PR
* [ ] Linted my code according to the [style guides](https://docs.simpeg.xyz/content/getting_started/contributing/code-style.html).
* [ ] Added [tests](https://docs.simpeg.xyz/content/getting_started/practices.html#testing) to verify changes to the code.
* [ ] Added necessary documentation to any new functions/classes following the
      expect [style](https://docs.simpeg.xyz/content/getting_started/practices.html#documentation).
* [ ] Marked as ready for review (if this is was a draft PR), and converted
      to a Pull Request
* [ ] Tagged ``@simpeg/simpeg-developers`` when ready for review.

#### Reference issue

Noticed the need to add these checks while reviewing #1390

### CODE REVIEW

File: SimPEG/electromagnetics/static/utils/static_utils.py [hunk 1]
```diff
@@ -356,20 +356,38 @@ def convert_survey_3d_to_2d_lines(
         A DC (or IP) survey
     lineID : (n_data) numpy.ndarray
         Defines the corresponding line ID for each datum
-    data_type : {'volt', 'apparent_resistivity', 'apparent_conductivity', 'apparent_chargeability'}
+    data_type : str, optional
```
Comment (jcapriot) [inline SimPEG/electromagnetics/static/utils/static_utils.py:359]: Leave this as it was, see https://numpydoc.readthedocs.io/en/latest/format.html#parameters. 
"When a parameter can only assume one of a fixed set of values, those values can be listed in braces, with the default appearing first:

```
order : {'C', 'F', 'A'}
    Description of `order`.
```

"

**Review by jcapriot — CHANGES_REQUESTED** [review]

File: SimPEG/electromagnetics/static/utils/static_utils.py (hunk 1 above)
Comment (santisoler) [reply SimPEG/electromagnetics/static/utils/static_utils.py:359]: Good to know. Thanks for the comment!

Comment (santisoler) [inline SimPEG/electromagnetics/static/utils/static_utils.py:359]:
```suggestion
    data_type : {'volt', 'apparent_resistivity', 'apparent_conductivity', 'apparent_chargeability'}
```

File: SimPEG/electromagnetics/static/utils/static_utils.py [hunk 2]
```diff
@@ -358,18 +358,36 @@ def convert_survey_3d_to_2d_lines(
         Defines the corresponding line ID for each datum
     data_type : {'volt', 'apparent_resistivity', 'apparent_conductivity', 'apparent_chargeability'}
         Data type for the survey.
-    output_indexing : bool, default=``False``
+        It can be one of: 'volt', 'apparent_resistivity',
+        'apparent_conductivity', and 'apparent_chargeability'.
```
Comment (jcapriot) [inline SimPEG/electromagnetics/static/utils/static_utils.py:362]: you also don't need these anymore

Comment (santisoler) [inline SimPEG/electromagnetics/static/utils/static_utils.py:362]:
```suggestion
```

File: simpeg/electromagnetics/static/utils/static_utils.py [hunk 3]
```diff
@@ -358,18 +358,34 @@ def convert_survey_3d_to_2d_lines(
         Defines the corresponding line ID for each datum
     data_type : {'volt', 'apparent_resistivity', 'apparent_conductivity', 'apparent_chargeability'}
         Data type for the survey.
-    output_indexing : bool, default=``False``
+    output_indexing : bool, default=``False``, optional
```
Comment (santisoler) [inline simpeg/electromagnetics/static/utils/static_utils.py:361]:
```suggestion
    output_indexing : bool, default=False, optional
```

**Pull request merged**